typings/text/encoder.d.ts

```diff
@@ -0,0 +1,38 @@
+/*
+* Copyright (c) 2022 Satoshi Tanaka
+*
+*   This file is part of the Moddable SDK Tools.
+*
+*   The Moddable SDK Tools is free software: you can redistribute it and/or modify
+*   it under the terms of the GNU General Public License as published by
+*   the Free Software Foundation, either version 3 of the License, or
+*   (at your option) any later version.
+*
+*   The Moddable SDK Tools is distributed in the hope that it will be useful,
+*   but WITHOUT ANY WARRANTY; without even the implied warranty of
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*   GNU General Public License for more details.
+*
+*   You should have received a copy of the GNU General Public License
+*   along with the Moddable SDK Tools.  If not, see <http://www.gnu.org/licenses/>.
+*
+*/
+
+declare module "text/encoder" {
+
+    interface TextEncoderEncodeIntoResult {
+        read?: number;
+        written?: number;
+    }
+    interface TextEncoder {
+        encode(input: string | number): Uint8Array;
```

Comment:
At first, I defined it as `string `,  but   `number` is set to `input` in [these tests](https://github.com/Moddable-OpenSource/moddable/blob/public/tests/modules/data/text/encoder/encode.js#L22-L23) .

Reply:
Good idea to check the tests. Note that the tests also pass a `BigInt`, which is different from a `number`.

The `encode` function expects a string. It coerces its input to a string. More-or-less, if something is not a string, then the `encode` calls `toString` on it. This is the same behavior as browsers implement for `encode` and the Web type declarations do not list ` number` as a type.

In TypeScript declarations, my understanding is an argument that is expected to be a string, even if other values are accepted by calling `toString`, is still declared as a `string` rather than enumerating all possible types that can be coerced (because that list is basically all types).